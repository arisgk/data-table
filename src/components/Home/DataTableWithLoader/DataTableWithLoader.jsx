import React from 'react';
import PropTypes from 'prop-types';
import FloatingActionButton from 'material-ui/FloatingActionButton';
import ContentAdd from 'material-ui/svg-icons/content/add';
import DataTable from './DataTable';
import Progress from './Progress';

const styles = {
  fab: {
    position: 'fixed',
    right: 20,
    bottom: 20,
  },
};

const DataTableWithLoader = ({ data: { loading, users } }) => {
  if (loading) {
    return (
      <Progress />
    );
  }

  return (
    <div>
      <DataTable users={users} />
      <FloatingActionButton style={styles.fab}>
        <ContentAdd />
      </FloatingActionButton>
    </div>
  );
};

DataTableWithLoader.propTypes = {
  data: PropTypes.shape({
    loading: PropTypes.bool,
    error: PropTypes.object,
    users: PropTypes.array,
  }),
};

DataTableWithLoader.defaultProps = {
  data: {
    loading: false,
    error: null,
    users: [],
  },
};

export default DataTableWithLoader;
