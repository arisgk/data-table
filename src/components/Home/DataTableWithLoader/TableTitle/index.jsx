import TableTitle from './TableTitle';

export default TableTitle;
