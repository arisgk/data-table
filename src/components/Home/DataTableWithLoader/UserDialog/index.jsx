import UserDialog from './UserDialog';

export default UserDialog;
