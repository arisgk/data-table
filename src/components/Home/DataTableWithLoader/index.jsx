import DataTableWithLoader from './DataTableWithLoader';

export default DataTableWithLoader;
