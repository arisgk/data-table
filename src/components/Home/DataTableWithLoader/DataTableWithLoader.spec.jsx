/* global describe, it, expect */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DataTableWithLoader from './DataTableWithLoader';

describe('DataTableWithLoader', () => {
  it('renders correctly', () => {
    const data = {
      loading: false,
      error: null,
      users: [],
    };

    let wrapper = shallow(
      <DataTableWithLoader data={data} />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();

    data.loading = true;
    wrapper = shallow(
      <DataTableWithLoader data={data} />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();

    data.loading = false;
    data.users = [
      {
        id: '1234abcd',
        firstName: 'Lebron',
        lastName: 'James',
        age: 32,
      },
      {
        id: '2345bcde',
        firstName: 'Steph',
        lastName: 'Curry',
        age: 29,
      },
    ];
    wrapper = shallow(
      <DataTableWithLoader data={data} />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
