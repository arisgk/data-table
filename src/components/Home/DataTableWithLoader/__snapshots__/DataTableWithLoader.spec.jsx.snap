// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableWithLoader renders correctly 1`] = `
<div>
  <TableTitle
    onStartDeleting={[Function]}
    selection={Object {}}
  />
  <DataTable
    users={Array []}
  />
  <FloatingActionButton
    disabled={false}
    mini={false}
    onClick={[Function]}
    secondary={false}
    style={
      Object {
        "bottom": 20,
        "position": "fixed",
        "right": 20,
      }
    }
    zDepth={2}
  >
    <ContentAdd />
  </FloatingActionButton>
  <UserDialog
    adding={false}
    listQuery={Object {}}
    mutate={[Function]}
    onCancel={[Function]}
  />
</div>
`;

exports[`DataTableWithLoader renders correctly 2`] = `<Progress />`;

exports[`DataTableWithLoader renders correctly 3`] = `
<div>
  <TableTitle
    onStartDeleting={[Function]}
    selection={Object {}}
  />
  <DataTable
    users={
      Array [
        Object {
          "age": 32,
          "firstName": "Lebron",
          "id": "1234abcd",
          "lastName": "James",
        },
        Object {
          "age": 29,
          "firstName": "Steph",
          "id": "2345bcde",
          "lastName": "Curry",
        },
      ]
    }
  />
  <FloatingActionButton
    disabled={false}
    mini={false}
    onClick={[Function]}
    secondary={false}
    style={
      Object {
        "bottom": 20,
        "position": "fixed",
        "right": 20,
      }
    }
    zDepth={2}
  >
    <ContentAdd />
  </FloatingActionButton>
  <UserDialog
    adding={false}
    listQuery={Object {}}
    mutate={[Function]}
    onCancel={[Function]}
  />
</div>
`;
